src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -395,11 +432,24 @@ public static BigDecimal constructDecimal(IonStruct constraintStruct) throws Exc
     }
 
     /**
-     * Generate random integers which composed by different length of data into the output file.
+     * Generate random integers which is conformed with the constraints provided in ISL.
+     * @param constraintStruct is an IonStruct which contains the top-level constraints in Ion Schema.
+     * @return constructed integers
+     * @throws Exception if error occurs when parsing the constraint 'valid_value'.
      */
-    public static long constructInt() {
+    public static long constructInt(IonStruct constraintStruct) throws Exception {
+        RandomDataGenerator randomDataGenerator = new RandomDataGenerator();
         Random random = new Random();
-        long longValue = random.nextLong();
+        long longValue;
+        if (IonSchemaUtilities.checkValidRange(constraintStruct) != null) {
+            IonList range = IonSchemaUtilities.checkValidRange(constraintStruct);
+            // Convert IonValue to long
+            long lowerBound = Long.valueOf(range.get(0).toString());
+            long upperBound = Long.valueOf(range.get(1).toString());
+            longValue = randomDataGenerator.nextLong(lowerBound, upperBound);
```

Comment:
We can achieve this behavior fairly easily without taking a dependency on `org.apache.commons.math3.random.RandomDataGenerator`. I recommend doing that unless there's another good reason for this dependency.

Reply:
> We can achieve this behavior fairly easily without taking a dependency on `org.apache.commons.math3.random.RandomDataGenerator`. I recommend doing that unless there's another good reason for this dependency.

Applied this suggestion in the following commit.